src/ring_theory/subsemiring.lean

```diff
@@ -369,6 +420,45 @@ lemma closure_eq_of_le {s : set R} {t : subsemiring R} (h₁ : s ⊆ t) (h₂ :
   closure s = t :=
 le_antisymm (closure_le.2 h₁) h₂
 
+end subsemiring
+
+namespace submonoid
+
+lemma to_subsemiring_coe :
+  (M.to_subsemiring : set R) = add_submonoid.closure (M : set R) := rfl
+
+end submonoid
+
+namespace subsemiring
+
+lemma to_subsemiring_to_add_submonoid :
+  M.to_subsemiring.to_add_submonoid = add_submonoid.closure (M : set R) := rfl
+
+/-- The `subsemiring.closure` of a multiplicative submonoid coincides with the `subsemiring`
+generated by the submonoid itself. -/
+lemma closure_eq_submonoid_to_subsemiring : subsemiring.closure (M : set R) = M.to_subsemiring :=
+begin
+  ext,
+  refine ⟨λ hx, (subsemiring.mem_closure.mp hx) M.to_subsemiring (λ s sM, _), λ hx, _⟩;
+  rintros - ⟨H1, rfl⟩;
+  rintros - ⟨H2, rfl⟩,
+  { exact H2 sM },
+  { exact add_submonoid.mem_closure.mp hx H1.to_add_submonoid H2 }
+end
+
+/-- The elements of the additive closure of a multiplicative submonoid `M` are exactly the
+elements of the subsemiring closure of `M`. -/
+lemma clss : (add_submonoid.closure (M : set R) : set R) = subsemiring.closure (M : set R) :=
+M.to_subsemiring_coe.symm.trans (by rw closure_eq_submonoid_to_subsemiring)
```

Comment:
I'm not really sure this lemma is worth having any more - it's hard to name and easy to prove.

Reply:
Ok, I removed this lemma!